### PR TITLE
Add RegexBasedEdgeLabelMapping and its tests

### PR DESCRIPTION
**Pull request by BentolhodaH**
Add RegexBasedEdgeLabelMapping and its tests.

### CODE REVIEW

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/RegexBasedEdgeLabelMappingToURIsImpl.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+
+public class RegexBasedEdgeLabelMappingToURIsImpl implements EdgeLabelMapping {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/RegexBasedEdgeLabelMappingToURIsImpl.java:6]: You can make this a subclass that extends `EdgeLabelMappingToURIsImpl`. Then, you don't need to repeat the implementation of the last two methods here, and the `map` method can be implemented to call `map` of the superclass.

File: src/test/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/RegexBasedEdgeLabelMappingToURIsImplTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import static org.junit.Assert.*;
+
+public class RegexBasedEdgeLabelMappingToURIsImplTest {
+    protected final String NSRELATIONSHIP = "https://example2.org/test/";
+    protected final String regex = "^\\w+";
+
+    protected final EdgeLabelMapping edgeLabelMapping = new RegexBasedEdgeLabelMappingToURIsImpl(regex, NSRELATIONSHIP);
+
+    @Test
+    public void mapEdgeLabel() {
```
Comment (hartig) [inline src/test/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/RegexBasedEdgeLabelMappingToURIsImplTest.java:16]: Can you please also include a test in which the given edge label does not match the regular expression and, thus, an exception is thrown.

**Review by hartig — CHANGES_REQUESTED** [review]

**Review by hartig — APPROVED** [review]
Looks all good now. Thanks!

I will merge this PR.

The next step will then be "SingleEdgeLabelMapping" or whatever we called it.
